openmc/model/surface_composite.py

```diff
@@ -52,6 +53,112 @@ def __neg__(self):
         """Return the negative half-space of the composite surface."""
 
 
+class CylinderSector(CompositeSurface):
+    """Infinite cylindrical sector composite surface.
+
+    A cylinder sector is composed of two cylindrical and two planar surfaces.
+    The cylindrical surfaces are concentric, and the planar surfaces intersect
+    the central axis of the cylindrical surfaces.
+
+    This class acts as a proper surface, meaning that unary `+` and `-`
+    operators applied to it will produce a half-space. The negative
+    side is defined to be the region inside of the cylinder sector.
+
+    Parameters
+    ----------
+    center : iterable of float
+       Coordinate for central axes of cylinders in the (y, z), (z, x), or (x, y)
+       basis. Defaults to (0,0).
+    r1 : float
+        Inner radius of sector. Must be less than r2.
+    r2 : float
+        Outer radius of sector. Must be greater than r1.
+    central_angle : float
+        Central angle, :math:`\\theta`, of the sector in degrees. Must be
+        greater that 0 and less than 360.
+    ccw_offset : float
+        Angular offset, :math:`\\alpha`, of the clockwise-most side of the
+        sector in degrees. The offset is in the counter-clockwise direction
+        with respect to the first basis axis (+y, +z, or +x). Note that
+        negative values translate to an offset in the clockwise direction.
```

Comment:
I actually find this to be a lot more confusing and had to look at the code just to understand what it means. I still think theta1 and theta2 parameters (along with a constraint that theta2 > theta1) would be the most intuitive thing for a user.

Reply:
Okay. I implemented the theta1, theta2 formulation. I still think it's useful to have an sector width-offset formulation, so I added a class function `from_theta_alpha` to allow users that option.